lib/queue_bus/config.rb

```diff
@@ -1,102 +1,70 @@
-module QueueBus
-  class Config
-    def adapter=val
-      raise "Adapter already set to #{@adapter_instance.class.name}" if @adapter_instance
-      if val.is_a?(Class)
-        @adapter_instance = name_or_klass.new
-      elsif val.is_a?(::QueueBus::Adapters::Base)
-        @adapter_instance = val
-      else
-        class_name = ::QueueBus::Util.classify(val)
-        @adapter_instance = ::QueueBus::Util.constantize("::QueueBus::Adapters::#{class_name}").new
-      end
-      @adapter_instance
-    end
-
-    def adapter
-      return @adapter_instance if @adapter_instance
-      raise "no adapter has been set"
-    end
-
-    def redis(&block)
-      # TODO: could allow setting for non-redis adapters
-      adapter.redis(&block)
-    end
+# frozen_string_literal: true
 
-    def default_app_key=val
-      @default_app_key = Application.normalize(val)
-    end
+require 'socket'
+require 'logger'
 
-    def default_app_key
-      @default_app_key
-    end
+module QueueBus
+  # This class contains all the configuration for a running queue bus application.
+  class Config
+    attr_accessor :default_queue, :local_mode, :hostname, :incoming_queue, :logger
```

Comment:
This change (plus initialize) does more for me to clean things up than the `has_adapter?` addition (though like both)